_posts/2016-08-03-better-topic-detection-in-text-with-lda2vec.md

```diff
@@ -0,0 +1,106 @@
+
+A second disadvantage is that it effectively sees a document as a bag of words, it doesn’t use context, how they are ordered and grouped together.
+
+### word2vec
+
+word2vec is a vector space model, in that it represents words as vectors with an arbitrary number of dimensions. All words are located in an x-dimensional vector space.
+The underlying idea of word2vec is that if certain words are close together (euclidean distance) in many documents, they probably have a lot in common in terms of semantics (this is called the "distributional hypothesis"). Google have written a <a href="https://papers.nips.cc/paper/5021-distributed-representations-of-words-and-phrases-and-their-compositionality.html" title="Original paper by Google researchers">paper</a> on it.
+
+It’s even the case that relationships in meaning are reflected in the vector space location, e.g. king - man + woman = queen (or country to capitals, as in the graph below)
+
+![Picture of a PCA Chart](/assets/images/pca-pic.png "PCA image from Google original paper")
+
+There are two main variants for word2vec, one is CBOW (continuous bag of words) which predicts a word from its context, and skip-gram which attempt to predict context words from its target. CBOW does better on small data sets, skip-gram better on large ones (<a href="https://href.li/?https://www.tensorflow.org/versions/r0.7/tutorials/word2vec/index.html">source here</a>).
+
+Skip-Gram models uses a moving window around every word (like n words before and n words after) to have a context, so that we have an input-output tuple as follows: e.g. "the quick brown fox jumped over the lazy dog" has ([the, brown], quick) as a (context, target) tuple.
+
+So in this case the objective function is to try to maximize the probability of "the" and "brown" given "fox" – for all words and all documents of the corpus.
+
+Skip-gram models with negative sampling add a negative term to remove ‘noise words’, that is words that are present in all contexts but don’t add much in terms of semantics (‘stop words’ like the, a, in, etc).
+
+All this is trained with feedforward neural networks (networks without cycles, straightforward input – projection – hidden – output layers).
+
+This is a computationally intensive process, so much work has been done in getting clever shortcuts.
+
+Improvements that were added in the model: using idiomatic phrases instead of words – some words should be grouped together, say the Financial Times has little to do with time measurement.
+
+As an interesting aside: Google has the software patent for this technique, but doesn't seem to have done much with that patent at the moment.
+
+## Lda2vec
+
+Lda2vec is a research project by Chris E. Moody, PhD at Caltech.  Lda2vec’s aim is to find topics while also learning word vectors to obtain sparser topic vectors that are easier to interpret, while also training the other words of the topic in the same vector space (using neighbouring words).
```

Comment:
probably need the link to https://github.com/cemoody/lda2vec here? (I ommitted it myself :/)